src/Language/PureScript/Pretty/Types.hs

```diff
@@ -99,6 +99,10 @@ kinded = mkPattern match
 insertPlaceholders :: Type -> Type
 insertPlaceholders = everywhereOnTypesTopDown convertForAlls . everywhereOnTypes convert
   where
+  -- Replacing applications of type level functions with placeholders
+  -- that we'll interpret later.
+  convert (TypeApp (TypeApp (TypeConstructor f) x) ret) | f == primName "TypeConcat" = PrettyPrintTypeApplication "TypeConcat" x ret
```

Comment:
I'd rather specialize to `PrettyPrintTypeConcat` or something, instead of putting a string in the type like this.